### PR TITLE
Heywood Healthcare

### PR DESCRIPTION
**Pull request by pfurbacher**
Updated Heywood Healthcare scraper, now based on node-fetch and node-html-parser. Tests use saved pages with and without slots. See issue #215.

### CODE REVIEW

**Review by harcod — CHANGES_REQUESTED** [review]
Please delete the "HeywoodHealthcare" directory that is in `site-scrapers` since you are now a `no-browser-site-scraper`.  Otherwise they will both be running!  (I tested the wrong one at first)

Also, if you merge with `origin/master`, the 'alerts' test has been fixed.

File: no-browser-site-scrapers/HeywoodHealthcare/config.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    public: {
+        name: "Heywood Healthcare",
+        street: "171 Kendall Pond Road W",
+        city: "Gardner",
+        state: "MA",
+        zip: "01440",
+        signUpLink: " https://gardnervaccinations.as.me/schedule.php",
+        extraData: {
+            Note:
+                "Vaccinations will be at the Polish American Citizens Club (PACC), 171 Kendall Pond Rd W, Gardner, MA 01440",
```
[the remark this answers is not in the document]
Comment (pfurbacher) [reply no-browser-site-scrapers/HeywoodHealthcare/config.js]: Done: 
1) touched up Note:
2) merged changes from upstream master
3) deleted original HeywoodHealthcare in site-scrapers

[the remark this answers is not in the document]
Comment (pfurbacher) [reply no-browser-site-scrapers/HeywoodHealthcare/config.js]: Oops. Forgot to delete the original test file.

Committed.

[the remark this answers is not in the document]
Comment (harcod) [reply no-browser-site-scrapers/HeywoodHealthcare/config.js]: Excellent!

File: test/HeywoodHealthcareTest.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    yield "firstPage-noSlots-hasMoreTimes-scripts-removed.html";
+    yield "secondPage-noSlots-scripts-removed.html";
+}
+const noAvailabilityFilenameGenerator = noAvailabilityFilenames();
+
+const noAvailabilityFetchService = {
+    async fetchAvailability(/*site*/) {
+        return loadTestHtmlFromFile(
+            "HeywoodHealthcare",
+            noAvailabilityFilenameGenerator.next().value
```
Comment (harcod) [inline test/HeywoodHealthcareTest.js:103]: `testFolderName` should be `"HeywoodHealthCare"`.  **(capitalize the C)**.
I'm running the tests on Linux which has case-sensitive filenames.

File: test/HeywoodHealthcareTest.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    yield "secondPage-lots0fSlots-scripts-removed.html";
+    yield "thirdPage-someSlots-noMoreTimes-scripts-removed.html";
+}
+const availabilityFilenameGenerator = availabilityFilenames();
+
+const availabilityFetchService = {
+    async fetchAvailability(/*site*/) {
+        return loadTestHtmlFromFile(
+            "HeywoodHealthcare",
+            availabilityFilenameGenerator.next().value
```
Comment (harcod) [inline test/HeywoodHealthcareTest.js:120]: `testFolderName` should be `"HeywoodHealthCare"`.  **(capitalize the C)**.

File: test/HeywoodHealthcareTest.js (hunk 2 above)
Comment (harcod) [reply test/HeywoodHealthcareTest.js:103]: Now I'm thinking that you should probably rename the directory instead.  The scraper is called "HeywoodHealthcare" (with only the H's capitalized), but the test folder is HeywoodHealthCare (with HHC capitalized).

Also, the real place calls itself "Heywood Healthcare"

**Review by harcod — CHANGES_REQUESTED** [review]
Overall suggestion is to rename the directory `test/HeywoodHealthCare` to be `test/HeywoodHealthcare` (with a lowercase c)

File: test/HeywoodHealthcareTest.js (hunk 3 above)
Comment (harcod) [reply test/HeywoodHealthcareTest.js:120]: same as above... renaming the directory will fix this too

Comment (pfurbacher) [reply test/HeywoodHealthcareTest.js:120]: Hmm, I just renamed the folder to HeywoodHealthcare in my working folder, but that doesn't seem to be sensed as a change by either VS Code or Git Desktop. Any suggestions as to how to get one or both to see the change? Or is there a command-line way of forcing this change?

File: test/HeywoodHealthcareTest.js (hunk 2 above)
Comment (pfurbacher) [reply test/HeywoodHealthcareTest.js:103]: Yeah, I thought I had eradicated the "C" in all references to the site after I noticed that they didn't capitalize it. Apparently the test resources folder evaded my notice.

File: test/HeywoodHealthcareTest.js (hunk 3 above)
Comment (pfurbacher) [reply test/HeywoodHealthcareTest.js:120]: Solved it by introducing a spelling change, committing, changing to the desired spelling and case, and committing again.

**Review by harcod — DISMISSED** [review]
Interesting fix for case changes in filenames.  It worked!

File: no-browser-site-scrapers/HeywoodHealthcare/config.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+const site = {
+    public: {
+        name: "Heywood Healthcare",
+        street: "171 Kendall Pond Road W",
+        city: "Gardner",
+        state: "MA",
+        zip: "01440",
+        signUpLink: " https://gardnervaccinations.as.me/schedule.php",
```
Comment (iann) [inline no-browser-site-scrapers/HeywoodHealthcare/config.js:8]: extra leading space in signuplink

**Review by iann — CHANGES_REQUESTED** [review]
Everything else looks good except the extra space, which may not matter to the UI, but it stuck out to me in the output json.

File: no-browser-site-scrapers/HeywoodHealthcare/config.js (hunk 4 above)
Comment (pfurbacher) [reply no-browser-site-scrapers/HeywoodHealthcare/config.js]: Thanks. Change committed.

**Review by harcod — APPROVED** [review]

**Pull request merged**